lib/ex_machina/ecto_strategy.ex

```diff
@@ -3,8 +3,14 @@ defmodule ExMachina.EctoStrategy do
 
   use ExMachina.Strategy, function_name: :insert
 
+  def handle_insert(%{__meta__: %{state: :loaded}} = record, _) do
+    raise ArgumentError,
+    "you called `insert` on #{inspect record} but it has already been saved.
```

Comment:
So I just realized this may not be ideal since the record could be *really big* so maybe something like

```elixir
you called `insert` on a record that has already been inserted # Note the wording changed to "been inserted"
You can only call `insert` on records that haven't been inserted yet.

The record you attempted to save:

#{inspect record, limit: :infinity}
```

That way the record comes afterward and doesn't appear in the middle of the sentence